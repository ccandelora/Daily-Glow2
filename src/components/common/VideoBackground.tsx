import React from 'react';
import { StyleSheet } from 'react-native';
import { ResizeMode, Video } from 'expo-av';

export const VideoBackground: React.FC = () => {
  return (
    <Video
      source={require('../../../assets/11985491_1080_1920_30fps.mp4')}
      style={styles.backgroundVideo}
      shouldPlay
      isLooping
      isMuted
      resizeMode={ResizeMode.COVER}
      useNativeControls={false}
    />
  );
};

const styles = StyleSheet.create({
  backgroundVideo: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
  },
}); 