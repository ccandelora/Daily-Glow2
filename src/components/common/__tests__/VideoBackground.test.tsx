import React from 'react';
import { render } from '@testing-library/react-native';
import { VideoBackground } from '../VideoBackground';

// Mock the expo-av Video component
jest.mock('expo-av', () => ({
  Video: 'Video',
  ResizeMode: {
    COVER: 'cover',
  },
}));

describe('VideoBackground', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<VideoBackground />);
    expect(toJSON()).toMatchSnapshot();
  });
}); 