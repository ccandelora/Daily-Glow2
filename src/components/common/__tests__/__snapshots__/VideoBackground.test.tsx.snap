// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoBackground renders correctly 1`] = `
<Video
  isLooping={true}
  isMuted={true}
  resizeMode="cover"
  shouldPlay={true}
  source={
    {
      "testUri": "../../../assets/11985491_1080_1920_30fps.mp4",
    }
  }
  style={
    {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  useNativeControls={false}
/>
`;
